### PR TITLE
feat(accordion): accordion component

### PR DESCRIPTION
**Pull request by pnagarjun16**
[Accordion Component](https://www.notion.so/rational-nomads/Accordion-component-ad54beb0dff64b7d9327af02e8a6f030)

## 1. Pull request checklist
- [x] Notion card has been moved to the Code Review column
- [x] Notion card has a link to this PR
- [x] A reviewer has been assigned to the Notion card

## 2. PR type
    - Feature

## 3. What is the old behaviour?
The accordion component is not available

## 4. What is the new behaviour?
Integrated the accordion component

### CODE REVIEW

File: src/components/Accordion/Accordion.module.scss [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+    &:hover {
+      cursor: pointer;
+    }
+  }
+
+  .AccordionTrigger {
+    font-family: inherit;
+    padding: var(--zui-accordion-item-padding);
+    // height: 36px;
```
Comment (domw30) [inline src/components/Accordion/Accordion.module.scss:122]: Lets remove any commented code :)

File: src/components/Accordion/Accordion.module.scss [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+      }
+
+      &:hover {
+        --zui-accordion-header-background-color: #{$purple-2};
+      }
+    }
+  }
+}
+
+.Accordion {
```
Comment (domw30) [inline src/components/Accordion/Accordion.module.scss:79]: I think we can simplify this file. Could we group styles together to improve readability. We should be able to declare classNames once and nest the styles

**Review by domw30 — CHANGES_REQUESTED** [review]
Looking good! I've requested a couple of small changes here. Also, I have asked the design team to provide a design review of the component. If there are any requested changes from design team I will share. Thank you!

File: src/components/Accordion/Accordion.module.scss (hunk 1 above)
Comment (pnagarjun16) [reply src/components/Accordion/Accordion.module.scss]: Removed

File: src/components/Accordion/Accordion.module.scss (hunk 2 above)
Comment (pnagarjun16) [reply src/components/Accordion/Accordion.module.scss:79]: As per our conversation, we keeping the variables.

File: src/components/Accordion/Accordion.module.scss [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    display: flex;
+    width: 100%;
+    border-bottom: var(--zui-accordion-border-bottom);
+
+    &:hover {
+      cursor: pointer;
+    }
+  }
+
+  .AccordionTrigger {
```
Comment (colbr) [inline src/components/Accordion/Accordion.module.scss:119]: You can remove `Accordion` from all of these child classes - CSS modules will scope them correctly by default so you can use simple names.

File: src/components/Accordion/Accordion.test.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    }
+  ]
+};
+
+beforeEach(() => {
+  jest.resetAllMocks();
+});
+
+describe('<Accordion />', () => {
+  test('Accordion must be in closed state by default', () => {
```
Comment (colbr) [inline src/components/Accordion/Accordion.test.tsx:20]: `should be in closed state by default`

File: src/components/Accordion/Accordion.test.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  });
+
+  test('Accordion must be in closed state when user clicks on accordion header two times', () => {
+    const { container } = render(<Accordion {...DEFAULT_PROPS} />);
+    const selectAccordionTriggerElement = container.getElementsByClassName('AccordionTrigger')[0];
+    fireEvent.click(selectAccordionTriggerElement);
+    expect(selectAccordionTriggerElement).toHaveAttribute('data-state', 'open');
+    fireEvent.click(selectAccordionTriggerElement);
+    expect(selectAccordionTriggerElement).toHaveAttribute('data-state', 'closed');
+  });
```
Comment (colbr) [inline src/components/Accordion/Accordion.test.tsx:40]: Could combine these two - `should toggle "open" state on click`.

```
test 'should toggle "open" state on click'
    // select a closed toggle
    // click it
    // check that it's now open
    // click it
    // check that it's now closed
```

File: src/components/Accordion/Accordion.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import { IconChevronDown } from '../Icons';
+
+import classNames from 'classnames';
+import styles from './Accordion.module.scss';
+
+const cx = classNames.bind(styles);
+
+export interface AccordionItem {
+  title: string;
+  text: string;
```
Comment (colbr) [inline src/components/Accordion/Accordion.tsx:13]: Can we change this to `content: ReactNode`? I imagine there will be times where we want to put links/buttons/etc in the text, so will need to be `ReactNode` to handle those cases.

File: src/components/Accordion/Accordion.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+            className={styles.AccordionItem}
+            value={`item-${i}`}
+            key={i}
+            data-focus-on-key-press={focusState}
+          >
+            <RadixAccordion.Header className={styles.AccordionHeader}>
+              <RadixAccordion.Trigger
+                className={styles.AccordionTrigger}
+                onMouseDown={() => setFocusState(false)}
+                onKeyDown={e => setFocusState(true)}
```
Comment (colbr) [inline src/components/Accordion/Accordion.tsx:36]: Remove the `e` if not using it, i.e. `() => setFocusState(true)`

File: src/components/Accordion/Accordion.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+export const Accordion = ({ items, contrast = 'high' }: AccordionProps) => {
+  const [focusState, setFocusState] = useState(false);
+  return (
+    <div className={styles.Accordion} data-contrast={contrast}>
+      <RadixAccordion.Root className={styles.AccordionRoot} type="single" collapsible>
+        {items.map((itm, i) => (
+          <RadixAccordion.Item
+            className={styles.AccordionItem}
+            value={`item-${i}`}
+            key={i}
```
Comment (colbr) [inline src/components/Accordion/Accordion.tsx:29]: Use `itm.title` as the key.

[Index as a key is an anti-pattern](https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/)

**Review by colbr — CHANGES_REQUESTED** [review]
Couple changes, looking good overall!

One design inaccuracy: padding should be applied on hover.

**Storybook**
Screen Shot 2023-02-08 at 5 40 48 PM

**Figma**
Screen Shot 2023-02-08 at 5 40 57 PM

**Comment by pnagarjun16** [comment]
> Couple changes, looking good overall!
> 
> One design inaccuracy: padding should be applied on hover.
> 
> **Storybook** Screen Shot 2023-02-08 at 5 40 48 PM
> 
> **Figma** Screen Shot 2023-02-08 at 5 40 57 PM

Initially, I added the Padding but Stacy asked me to remove it.

File: src/components/Accordion/Accordion.module.scss (hunk 3 above)
Comment (pnagarjun16) [reply src/components/Accordion/Accordion.module.scss:119]: Updated

File: src/components/Accordion/Accordion.test.tsx (hunk 4 above)
Comment (pnagarjun16) [reply src/components/Accordion/Accordion.test.tsx:20]: Updated

File: src/components/Accordion/Accordion.test.tsx (hunk 5 above)
Comment (pnagarjun16) [reply src/components/Accordion/Accordion.test.tsx:40]: Updated

File: src/components/Accordion/Accordion.tsx (hunk 6 above)
Comment (pnagarjun16) [reply src/components/Accordion/Accordion.tsx:13]: Updated to ReactNode

File: src/components/Accordion/Accordion.tsx (hunk 7 above)
Comment (pnagarjun16) [reply src/components/Accordion/Accordion.tsx:36]: removed.

File: src/components/Accordion/Accordion.tsx (hunk 8 above)
Comment (pnagarjun16) [reply src/components/Accordion/Accordion.tsx:29]: The title might be the same so I added it like ```key={`${itm.title}-${i}`}```

**Review by colbr — DISMISSED** [review]

**Pull request merged**